src/compiler/mixin.cc

```diff
@@ -23,6 +23,171 @@
 namespace toit {
 namespace compiler {
 
+/// A visitor that finds the actual super call.
+/// The 'Super' class contains the full expression that is
+/// responsible for a super call. This could have been compiled
+/// to a sequence, if some of the arguments had to be hoisted.
+/// This visitor finds the 'CallStatic' of the constructor call.
+class SuperCallVisitor : protected ir::ReplacingVisitor {
+ public:
+  explicit SuperCallVisitor(ir::Class* holder) : super_(holder->super()) {}
+
+  bool has_seen_super() const { return has_seen_super_; }
+
+  // Make `visit` public.
+  ir::Node* visit(ir::Node* node) override {
+    return ReplacingVisitor::visit(node);
+  }
+
+ protected:
+  ir::Node* visit_Super(ir::Super* node) override {
+    if (node->expression() == null) {
+      auto replacement = visit_static_super_call(null, node->range());
+      if (replacement != null) {
+        node->replace_expression(replacement->as_Expression());
+      }
+      return node;
+    }
+    in_super_ = true;
+    auto result = ir::ReplacingVisitor::visit_Super(node);
+    in_super_ = false;
+    has_seen_super_ = true;
+    return result;
+  }
+
+  ir::Node* visit_CallStatic(ir::CallStatic* node) override {
+    if (!in_super_) return ir::ReplacingVisitor::visit_CallStatic(node);
+    auto method = node->target()->target();
+    if (method->is_constructor() && method->holder() == super_) {
```

Comment:
I guess a static call here that isn't to the super constructor is part of an argument? Maybe explain this.